lib/uring/uring.mli

```diff
@@ -67,6 +67,14 @@ val noop : 'a t -> 'a -> 'a job option
 (** [noop t d] submits a no-op operation to uring [t]. The user data [d] will be
     returned by {!wait} or {!peek} upon completion. *)
 
+val timeout: 'a t -> ?clock:[`Clock_mono | `Clock_sys | `Clock_default] -> int64 -> 'a -> 'a job option
```

Comment:
This is a Linux-only library (not a cross-platform abstraction), so the names should match the ones in the C API:
```suggestion
val timeout: 'a t -> clock:[`Boottime|`Realtime] -> int64 -> 'a -> 'a job option
```